### ISSUE TITLE
CudaHalfTensor Feature Request

### ISSUE THREAD

**Issue by dgcrouse** [torch/cunn#315]
With the advent of Pascal's improved Half-Precision support, many of us would appreciate adding full CudaHalfTensor support to cunn. I am adding this as a feature request to help track progress and to help point devs who are interested in contributing to this (such as myself) toward where their contributions would be most valuable.

**Comment by borisfom** [comment]
@soumith, @szagoruyko : is there any work in progress on that?

**Comment by dgcrouse** [comment]
Additional note - Looks like half precision support is barely present (half speed of FP64) on all non-GP100 Pascal boards.

**Comment by borisfom** [comment]
Yes, as of right now you need GP100 to get speed advantage of fp16 math.

**Comment by soumith** [comment]
@borisfom cunn hasn't been started to be templatized. We've almost finished cutorch's multiple types work.
If you think you have any spare cycles, would love to see modules slowly converted (via the same way that we have nn and cutorch templatized, via the generic/ folder and using C macros.

**Comment by borisfom** [comment]
@soumith: thanks for the update! Sorry I won't have too many cycles soon for cunn work...

**Comment by soumith** [comment]
@gchanan has started working on this, an issue is here to track progress: https://github.com/torch/cunn/issues/347
We've already come quite far.
